core/src/autogluon/core/utils/feature_selection.py

```diff
@@ -0,0 +1,537 @@
+            self._debug_info['exceptions'].append(str(e))
+            if self.raise_exception:
+                raise e
+
+        if auto_threshold:
+            best_info['features'] = [feature for feature in best_info['features'] if self.noise_prefix not in feature]
+        if not self.keep_models and best_info['model'].name != self.original_model.name:
+            best_info['model'].delete_from_disk(silent=True)
+        self._debug_info['total_prune_time'] = self.original_time_limit - self.time_limit
+        self._debug_info['total_prune_fit_time'] = self._fit_time_elapsed
+        self._debug_info['total_prune_fi_time'] = self._fi_time_elapsed
+        self._debug_info['kept_ratio'] = len(best_info['features']) / len(original_features)
+        total_time_elapsed = round(self.original_time_limit - self.time_limit, 2)
+        logger.log(30, f"\tSuccessfully ended prune loop after {index} feature pruning rounds ({total_time_elapsed}s). Best score: {best_info['score']}.")
+        logger.log(30, f"\tFeature count before/after feature pruning: {len(original_features)} -> {len(best_info['features'])}.")
+        return best_info['features']
+
+    def compute_next_candidate(self, fn_args: dict, round_time_budget: float, prev_candidate_features: List[str], prev_importance_df: pd.DataFrame,
+                               ) -> Tuple[List[str], pd.DataFrame, bool, float]:
+        """
+        While time allows, repeatedly compute feature importance and generate candidate feature subsets using a fixed time budget.
+        If at least one feature can be pruned, return. If no feature is immediately pruned but time remains and some feature's
+        importance scores are not calculated, repeat the procedure. Note: If we have feature importance information from previously
+        fitted model, make use of them since we might not have time this iteration to evaluate importance scores for all features.
+        Any feature importance calculated within this call of compute_next_candidate will override previous feature importance info.
+        """
+        candidate_features = prev_candidate_features
+        importance_df = unevaluated_fi_df_template(candidate_features)
+        candidate_found = False
+        total_prune_time = 0.
+        # mark previous fit's computed feature importances here.
+        if prev_importance_df is not None:
+            fn_args['prev_importance_df'] = prev_importance_df
+            fn_args['using_prev_fit_fi'] = set(prev_importance_df[prev_importance_df['n'] > 0].index.tolist())
+        while self.time_limit > round_time_budget + self.model_fit_time:
+            candidate_features, importance_df, prune_time = self.compute_next_candidate_round(**fn_args)
+            # HACK: Line below is needed to get this working with n-repeated bagged models. Related to feature ordering.
+            candidate_features = [feature for feature in fn_args['X'].columns.tolist() if feature in candidate_features]
```

Comment:
Why exactly? Is this a bug in AG?

Reply:
This used to cause an issue for certain models when used with repeated bagging (ex. Catboost) related to feature ordering but I will test if it still is an issue after my changes.